ConsoleApp6/PaymentServices.cs

```diff
@@ -15,35 +18,62 @@ public PaymentServices(Products p)
             this.UPC = p.UPC;
             this.Tax = Products.Tax;
             this.Discount = Products.Discount;
+            this.specialUPC = Products.specialUPC;
+            this.upcDiscount = Products.UPC_Discount;
          
         }
-        public double addTax()
+        
+        double AddTax()
         {
 
-            double priceWithTax = Price + (Price * Tax / 100);
+            double priceWithTax = Price+(Price * Tax / 100);
             return Math.Round(priceWithTax, 2);
         }
 
-        public double createDiscount()
+        double CreateUniDiscount()
         {
-            double pricewithDiscount = addTax() - (Price * Discount / 100);
-            return Math.Round(pricewithDiscount, 2);
+            double amountofDiscount =Price * Discount / 100;
+            return Math.Round(amountofDiscount, 2);
         }
+        double CreateUPCDiscount()
+        {
+            double UPCDiscount=0;
+            if (specialUPC.Contains(UPC))
+                UPCDiscount = Price * upcDiscount / 100;
 
-        public void Report()
+            return Math.Round(UPCDiscount, 2);
+        }
+
+        void Report()
         {
             if (Discount != 0)
             {
-                Console.Write($"Tax Amount= ${Price * Tax / 100}, Discount Amount = ${Price * Discount / 100}" +
-                            $"\nTittle = {Name}, UPC = {UPC}, Price = ${createDiscount()}");
+                Console.WriteLine($"Tax Amount= ${Math.Round(Price * Tax / 100, 2)}, Discount Amount = ${CreateUniDiscount()}" +
+                            $"\nTittle = {Name}, UPC = {UPC}, Price = ${AddTax()-CreateUniDiscount()}");
 
             }
             else
             {
-                Console.Write($"Tax Amount = ${Price * Tax / 100}, no Discount\nTittle = {Name}, UPC = {UPC}, Price = ${addTax()}");
+                Console.WriteLine($"Tax Amount = ${Math.Round(Price * Tax / 100,2)}, no Discount\nTittle = {Name}, UPC = {UPC}, Price = ${AddTax()}");
             }
         }
 
 
+
+        public void Precedence(bool upcFlag,bool uniFlag) {
+            if (upcFlag && uniFlag)//apply  2 type of discount before tax 
+            {
+                Price = Price - (CreateUniDiscount() + CreateUPCDiscount());
```

Comment:
the method name should tell us what the method does, so it should be a verb, it's better to have one method that calculates the final price, which checks what discount details we have, and just calls other methods to compute the different rates,